This patch will fix wrong behavior on ghc-7.4.
See http://hackage.haskell.org/trac/ghc/ticket/4958
Tested on: ghc-7.4.1 and ghc-7.0.4

diff --git a/Main.hs b/Main.hs
index 25ed784..aaf76f8 100644
--- a/Main.hs
+++ b/Main.hs
@@ -1,4 +1,3 @@
-{-# LANGUAGE CPP #-}
 module Main ( main ) where
 
 --import qualified Data.ByteString as B
@@ -8,12 +7,7 @@ import qualified Text.Regex.PCRE.Light.Char8 as R
 import Data.List as L
 import Control.Applicative
 import Control.Monad
-
-#if __GLASGOW_HASKELL__>=704
-import System.Environment
-#else
 import System
-#endif
 import System.Directory
 import System.FilePath ( splitDirectories )
 import System.IO.Unsafe ( unsafeInterleaveIO )
diff --git a/keyword-stat.cabal b/keyword-stat.cabal
index 7564aa6..00baf81 100644
--- a/keyword-stat.cabal
+++ b/keyword-stat.cabal
@@ -7,12 +7,9 @@ license:             BSD3
 -- license-file:        LICENSE
 author:              Lennart Kolmodin
 maintainer:          Lennart Kolmodin <kolmodin@gentoo.org>
+build-Depends:       base, directory, haskell98, pcre-light, filepath
 build-type:          Simple
-cabal-version:       >= 1.2
 
-executable          keyword-stat
-  build-depends:       base,directory, pcre-light, filepath
-  if impl( ghc < 7.4)
-      build-depends: haskell98
-  main-is:             Main.hs
-  ghc-options:         -Wall
+executable:          keyword-stat
+main-is:             Main.hs
+ghc-options:         -Wall
